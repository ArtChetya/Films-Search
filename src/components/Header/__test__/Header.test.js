import { shallow } from 'enzyme/build';
import React from 'react';
import { Header } from '..';

describe('Header component', () => {
    it('should be rendered correctly without props', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });

    it('should be rendered correctly with hasSearch prop', () => {
        const component = shallow(<Header hasSearch />);
        expect(component).toMatchSnapshot();
    });

    it('should be rendered correctly with content prop', () => {
        const component = shallow(<Header content={<div>Test</div>} />);
        expect(component).toMatchSnapshot();
    });

    it('should be rendered correctly with hasSearch and content prop', () => {
        const component = shallow(
            <Header hasSearch content={<div>Test</div>} />
        );
        expect(component).toMatchSnapshot();
    });
});
