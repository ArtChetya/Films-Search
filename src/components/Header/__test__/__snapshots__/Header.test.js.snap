// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be rendered correctly with content prop 1`] = `
<AppBar
  bgcolor="#212121"
  minheight="36px"
  opacity={0.9}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    justify="space-between"
  >
    <WithStyles(Grid)
      item={true}
    >
      <ColorText
        fontcolor="#ec407a"
        variant="h6"
      >
        netflixroulette
      </ColorText>
    </WithStyles(Grid)>
    <Header__GridMarginTop
      item={true}
      xs={12}
    >
      <div>
        Test
      </div>
    </Header__GridMarginTop>
  </WithStyles(Grid)>
</AppBar>
`;

exports[`Header component should be rendered correctly with hasSearch and content prop 1`] = `
<AppBar
  bgcolor="#212121"
  minheight="36px"
  opacity={0.9}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    justify="space-between"
  >
    <WithStyles(Grid)
      item={true}
    >
      <ColorText
        fontcolor="#ec407a"
        variant="h6"
      >
        netflixroulette
      </ColorText>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <Link
        to="/"
      >
        <ColorButton
          size="medium"
          variant="contained"
        >
          Search
        </ColorButton>
      </Link>
    </WithStyles(Grid)>
    <Header__GridMarginTop
      item={true}
      xs={12}
    >
      <div>
        Test
      </div>
    </Header__GridMarginTop>
  </WithStyles(Grid)>
</AppBar>
`;

exports[`Header component should be rendered correctly with hasSearch prop 1`] = `
<AppBar
  bgcolor="#212121"
  minheight="36px"
  opacity={0.9}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    justify="space-between"
  >
    <WithStyles(Grid)
      item={true}
    >
      <ColorText
        fontcolor="#ec407a"
        variant="h6"
      >
        netflixroulette
      </ColorText>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
    >
      <Link
        to="/"
      >
        <ColorButton
          size="medium"
          variant="contained"
        >
          Search
        </ColorButton>
      </Link>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</AppBar>
`;

exports[`Header component should be rendered correctly without props 1`] = `
<AppBar
  bgcolor="#212121"
  minheight="36px"
  opacity={0.9}
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    justify="space-between"
  >
    <WithStyles(Grid)
      item={true}
    >
      <ColorText
        fontcolor="#ec407a"
        variant="h6"
      >
        netflixroulette
      </ColorText>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</AppBar>
`;
